test/test_internet.py

```diff
@@ -13,38 +13,46 @@
 
 
 class TestDatasetUtils:
-    def test_download_url(self, tmpdir):
+    @pytest.mark.parametrize("temp_dir", ("tmpdir", "tmp_path"))
+    def test_download_url(self, temp_dir, request):
+        temp_dir = request.getfixturevalue(temp_dir)
```

Comment:
Doing it this way works, but the main drawback is that it doesn't make the code very explicit about what we're trying to do. That is, our intent here is to test support for `pathlib.Path()`, but it's not obvious from the code that i) this is the actual intent behind using `tmp_path` and ii) that `pathlib.Path()` is being used at-all: one needs to know that the `tmp_path` fixture returns a `Path()`, and quite frankly... we don't know that from the top of our head. And even if we know it now, we'll forget very soon, and we'll look at these tests 3 months from now and wonder why we bothered testing these 2 fixtures.

So I'd recommend a more obvious alternative that clarifies the intent, e.g. something that you already suggested in our chat:

```py
@pytest.mark.parametrize("use_pathlib", (True, False))
# ...
    if use_pathlib:
        tmpdir = Path(tmpdir)
```

We have a strong bias towards simplicity in torchvision and in the domain libraries in general, for the sake of long-term maintainability. So we try hard to keep things dumb-simply and **very** obvious, even if sometimes that means a little bit of repetition/boilerplate (in this case, it does not).